### ISSUE TITLE
CollectionAssert.Empty enumerator not disposed

### ISSUE THREAD

**Issue by riezebosch** [xunit/xunit#1067]
https://github.com/xunit/assert.xunit/blob/431f1aaab702b2a184dd03ec56d7a565c6e916c8/CollectionAsserts.cs#L225

This results in an exception when used against a query from EntityFramework Core because the reader is not properly closed.

```cs
 Assert.NotEmpty(context.Person);
 Assert.NotEmpty(context.Person);
```

Output:
```
Test Outcome:	Failed
Test Duration:	0:00:01.627

Result StackTrace:	
at System.Data.SqlClient.SqlInternalConnectionTds.ValidateConnectionForExecute(SqlCommand command)
   at System.Data.SqlClient.SqlCommand.ValidateCommand(Boolean async, String method)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean asyncWrite, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, String executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean closeConnection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable.Enumerator.BufferlessMoveNext(Boolean buffer)
   at Microsoft.EntityFrameworkCore.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](Func`2 operation, Func`2 verifySucceeded, TState state)
   at Microsoft.EntityFrameworkCore.Query.QueryMethodProvider.<_ShapedQuery>d__3`1.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.<_TrackEntities>d__15`2.MoveNext()
   at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
   at Xunit.Assert.NotEmpty(IEnumerable collection) in C:\BuildAgent\work\cb37e9acf085d108\src\xunit.assert\Asserts\CollectionAsserts.cs:line 225
   at <my test file>
Result Message:	There is already an open DataReader associated with this Command which must be closed first.
```

Granted, I probably should force query execution first and assert on the in-memory list. But still this isn't desired behavior.

One options is to use the Any extension method from Linq or look at its [implementation](https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Linq/src/System/Linq/AnyAll.cs#L18)

**Comment by KirillShlenskiy** [comment]
Same problem in `CollectionAssert.Empty(IEnumerable)` too.

**Comment by bradwilson** [comment]
Fixed in 2.2 RC1.
